packages/cli/templates/defaults/app/redirects.ts

```diff
@@ -0,0 +1,28 @@
+import { matchRoutes } from "@remix-run/router";
+import { type PageRedirect } from "@webstudio-is/sdk";
+
+export const matchPathFromRedirects = (
+  url: string,
+  redirects: Array<PageRedirect>
+) => {
+  const redirectMap: Record<string, string> = {};
+  const routePaths: Array<{ path: string }> = [];
+
+  for (const redirect of redirects) {
+    redirectMap[redirect.old] = redirect.new;
+    routePaths.push({ path: redirect.old });
+  }
+
+  const matchedRoutes = matchRoutes(routePaths, new URL(url));
```

Comment:
Is remix able to match such routes syntax `/:id`?

Reply:
Yes, for the incoming routes. Let's say uses request with `/blog/1234`, `/blog/124234` etc. But while defining the redirects user might have just added `/blog/:id`. Then we need to reply on this. to match `/blog/1234` -> `/blog/:id` and then find the destination for the result.